### ISSUE TITLE
Configure Build & Deploy Ansible Roles and Jenkins Jobs

### ISSUE THREAD

**Issue by ElijahLynn** [department-of-veterans-affairs/va.gov-cms#3315]
Goal is to ship MVP of Tugboat and port PRs to it right now, but not DEMO environments, just yet. 

AC:
   - [ ] Enough disk space for 100 environments and two volumes: 
     - [ ] root is 20GB
     - [ ] Consider IO1 @ 3,000 IOPS instead of GP2 which bursts to 3,000 IOPS with credit limit
     - [ ] Raw 1000GB block device - unmounted (but attached) and unpartitioned & unformatted volume
   - [ ] Enough CPU and memory resources to build and run tests on 10 parallel environments running at the same time
   - [ ] Build is automated & triggerable via a Jenkins job
   - [ ] Deploy is automated & triggerable via a Jenkins job
   - [ ] Wrapper job to trigger build and deploy jobs

TASKS:
  - [ ] Tracking in public https://checkvist.com/p/vm10fSXd8WfU8pSFmSKoTN because it is lightyears faster, can't share a direct sub-link but here is the public breadcrumb to follow:  
  VA.gov V3 CMS Infrastructure  > Continuous Integration (CI) > Phase 1 - MVP - 2020.12.04
issues/2680 [SUPEREPIC] V3-Phase1 of CI/PR/Automation Testing/Demo/Local Environments · Issue #2680 > issues/3314 [EPIC] BRDize the Tugboat (cms-ci app) · Issue #3314

**Comment by ElijahLynn** [comment]
We are putting a manually configured instance up for initial roll-out and will be able to migrate it to the BRD system easily.

**Comment by ElijahLynn** [comment]
Here is what we ended up doing for the manual setup:

Manually configure Tugboat server
   - BRD Server Steps
      - Copy deploy key up, or credstash pull it
         - `yum install python-pip`
         - /root/.ssh/id_ed25519
         - `chmod 600 /root/.ssh/id_ed25519`
      - `git clone git@github.com:Lullabot/tugboat.git /opt/tugboat`
      - `git checkout v2.20.72`
      - Volumes/Partitions/Filesystems/Mounts
         - Ran commands in instances-inner.yml to create partition and make ext4 file system
         - Attach volumes (BRD does this though, so just a manual thing for now, not Ansible)
         - Partition and format docker-data volume (BRD does this)
         - Mount /dev/sdb to /opt/tugboat/data (BRD does this)
            - recreate the directory structure, including .gitignore files
      - yum install docker
         - update and copy daemon.json, see tugboat install
            - /dev/sdg
            - 1024G
      - service docker start
      - docker swarm init --advertise-addr 172.31.1.130
      - `make`
      - `make install`
      - Symlink bins from `make install`
         - ln -s /usr/local/bin/tugboat /usr/bin
         - ln -s /usr/local/sbin/tbctl /usr/sbin
      - cp /opt/tugboat/config.example.js /opt/tugboat/config.js
      - vim /opt/tugboat/config.js
         - config.key
         - config.secret
         - config.domain = tugboat.vfs.va.gov
         - config.email_authentication
         - config.session_secret
         - config.providers github
         - config.support.slack = https://dsva.slack.com/archives/C01A35JDH88
         - commit final changes to a branch/PR
      - `tbctl start`
      - Update project system resource variables with mongo shell (see Slack)
         - 36 vCPU
         - 10GB memory, should be 16GB
   - Fix webhook route
   - Fix dashboard generated link
   - Added cronjob to auto add users
   - UI Config Steps
      - Create cms-test project
         - Add repo twice, cms and web
         - set `tugboat domain <help>` per repo
            - demo.cms.va.gov
            - ci.cms.va.gov
      - Login with `va-cms-bot` user, as first user
      - Login with elijah, and neil
      - Make elijah, eric and neil admin users, mongo shell
   - tugboat update 5fd010a223d4f83f36366150 cpus=36 build_memory=72000 memory=72000 quota=1024

**Comment by ElijahLynn** [comment]
We ended up doing a manual install for now but did do a lot of the BRD part, just couldn't finish in time for the deadline so were pragmatic about shipping the value to the team. The manual story got moved to https://github.com/department-of-veterans-affairs/va.gov-cms/issues/3315#issuecomment-745051963 and I added the Sprint 21 milestone to it and removed it from here. Will complete these when I am back from vacation.
